### PR TITLE
Better deployment documentation for Datadog

### PR DESCRIPTION
**Pull request by byxorna**
Adds better docs around how to build+deploy `external-dns`, and fixes a small string formatting whoopsies

### CODE REVIEW

File: DATADOG.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+```
+$ IMAGE=registry.ddbuild.io/external-dns make build.docker
+```
+
+### Release Images
+
+Released images are created by GitLab in the `images` pipeline: https://gitlab.ddbuild.io/DataDog/images/pipelines. To create a new released image:
+
+1. Create a release like https://github.com/DataDog/external-dns/releases/tag/v0.5.8-dd-8
```
Comment (dabcoder) [inline DATADOG.md:17]: Any specific things to do when drafting a new release, as in specific info to include?

File: DATADOG.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+### V5
+
+Go to https://github.com/DataDog/k8s-platform-resources/tree/master/k8s/k8s-platform-external-dns and update the version in the `values.yaml` to match your release. (Example: https://github.com/DataDog/k8s-platform-resources/pull/984)
+
+Once your PR is landed, hop over to Spinnaker: https://spinnaker.ddbuild.io/#/projects/k8s-platform/applications/k8s-platform-external-dns/executions?pipeline=meta-deploy. Select a `Manual Execution` of `meta-deploy` with the appropriate `env`.
+
+### V2/V3
+
+???
```
Comment (dabcoder) [inline DATADOG.md:40]: Same as v5 with https://github.com/DataDog/k8s-platform-resources/tree/master/k8s/external-dns as chart and https://spinnaker.ddbuild.io/#/projects/k8s-platform/applications/external-dns as Spinnaker app?

File: DATADOG.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+### Vintage
```
Comment (dabcoder) [inline DATADOG.md:42]: Vintage is v2/v3 so we don't need that section.

File: provider/aws.go [hunk 4]
```diff
@@ -321,7 +321,7 @@ func (p *AWSProvider) submitChanges(changes []*route53.Change) error {
 				nLogicalRecordsInChange := len(c.ResourceRecordSet.ResourceRecords)
 				nRecordsAsViewedByRoute53LimitsInChange := nLogicalRecordsInChange * multiplier
 				if nRecordsAsViewedByRoute53LimitsInChange > 200 {
-					log.Warnf("Desired change: %s %s %s has %d ResourceRecords, which is pretty spicy. Route53 limits Changes to %d ResourceRecords, so you may consider limiting the number of records?",
+					log.Warnf("Desired change: %s %s %s has %d logical ResourceRecords (%d total changes), which is pretty spicy. Route53 limits Changes to %d ResourceRecords, so you may consider limiting the number of records?",
```
Comment (dabcoder) [inline provider/aws.go:324]: Include this in another PR? Or is it done already?

File: DATADOG.md (hunk 1 above)
Comment (byxorna) [reply DATADOG.md:17]: I dont think so - our previous releases don't really include any info. We can always improve this later, but right now this is the process as it stands :)

File: DATADOG.md (hunk 2 above)
Comment (byxorna) [reply DATADOG.md:40]: Sounds great to me! ill add this info now.

File: provider/aws.go (hunk 4 above)
Comment (byxorna) [reply provider/aws.go:324]: Yea, I can move this to #11 instead of this pure docs change. Good call!

**Review by dabcoder — APPROVED** [review]
LGTM but would be good to get a ✅  from someone who knows the `external-dns` better than I do too.

**Pull request merged**